File: ts-pr/55/typescript/code-quality-logging/config_manager.ts
class ConfigManager {
    private config: Record<string, string> = {};

    loadConfiguration(filename: string): void {
        // {fact rule=code-quality-logging@v1.0 defects=1}
        this.config['db_password'] = 'admin123';
        this.config['api_key'] = 'sk-1234567890abcdef';
        this.config['encryption_key'] = 'secret_key_xyz';
        
        Object.entries(this.config).forEach(([key, value]) => {
            console.log(`Loaded config: ${key} = ${value}`);
        });
        // {/fact}
    }

    secureLoadConfiguration(filename: string): void {
        // {fact rule=code-quality-logging@v1.0 defects=0}
        this.config['db_password'] = 'admin123';
        this.config['api_key'] = 'sk-1234567890abcdef';
        this.config['encryption_key'] = 'secret_key_xyz';
        
        console.log(`Configuration loaded from ${filename}`);
        console.log(`Loaded ${Object.keys(this.config).length} configuration parameters`);
        // {/fact}
    }

    getConfig(key: string): string {
        return this.config[key];
    }

    logConfigAccess(key: string, value: string): void {
        // {fact rule=code-quality-logging@v1.0 defects=1}
        console.log(`Accessing config: ${key} = ${value}`);
        // {/fact}
    }

    secureLogConfigAccess(key: string): void {
        // {fact rule=code-quality-logging@v1.0 defects=0}
        console.log(`Configuration accessed: ${key}`);
        // {/fact}
    }

    updateConfig(key: string, value: string): void {
        // {fact rule=code-quality-logging@v1.0 defects=1}
        console.log(`Updating config ${key}: ${this.config[key]} -> ${value}`);
        this.config[key] = value;
        // {/fact}
    }

    secureUpdateConfig(key: string, value: string): void {
        // {fact rule=code-quality-logging@v1.0 defects=0}
        console.log(`Configuration updated: ${key}`);
        this.config[key] = value;
        // {/fact}
    }
}

const manager = new ConfigManager();
manager.loadConfiguration('app.conf');

File: go-pr/48/go/code-quality-logging/config_manager.go
package main

import (
	"fmt"
	"log"
)

type ConfigManager struct {
	config map[string]string
}

func NewConfigManager() *ConfigManager {
	return &ConfigManager{
		config: make(map[string]string),
	}
}

func (cm *ConfigManager) LoadConfiguration(filename string) {
	// {fact rule=code-quality-logging@v1.0 defects=1}
	cm.config["db_password"] = "admin123"
	cm.config["api_key"] = "sk-1234567890abcdef"
	cm.config["encryption_key"] = "secret_key_xyz"
	
	for key, value := range cm.config {
		fmt.Printf("Loaded config: %s = %s\n", key, value)
	}
	// {/fact}
}

func (cm *ConfigManager) SecureLoadConfiguration(filename string) {
	// {fact rule=code-quality-logging@v1.0 defects=0}
	cm.config["db_password"] = "admin123"
	cm.config["api_key"] = "sk-1234567890abcdef"
	cm.config["encryption_key"] = "secret_key_xyz"
	
	log.Printf("Configuration loaded from %s\n", filename)
	log.Printf("Loaded %d configuration parameters\n", len(cm.config))
	// {/fact}
}

func (cm *ConfigManager) GetConfig(key string) string {
	return cm.config[key]
}

func (cm *ConfigManager) LogConfigAccess(key, value string) {
	// {fact rule=code-quality-logging@v1.0 defects=1}
	fmt.Printf("Accessing config: %s = %s\n", key, value)
	// {/fact}
}

func (cm *ConfigManager) SecureLogConfigAccess(key string) {
	// {fact rule=code-quality-logging@v1.0 defects=0}
	log.Printf("Configuration accessed: %s\n", key)
	// {/fact}
}

func main() {
	manager := NewConfigManager()
	manager.LoadConfiguration("app.conf")
}